arangod/Cluster/ClusterInfo.h

```diff
@@ -917,9 +971,11 @@ class ClusterInfo final  {
 
   uint64_t _planVersion;     // This is the version in the Plan which underlies
                              // the data in _plannedCollections and _shards
+  uint64_t _planIndex;       // This is the Raft index, which corresponds to the above plan version
```

Comment:
We probably want to adjust the comments around `ProtectionData`. I wonder what the actual meaning of the mutex now is, probably it is only for protecting `_currentIndex` and `_planIndex` for the `waitFor` call. `wantedVersion` and `doneVersion` can probably be removed. The ReadWriteLock is still used in the same way, right? Please adjust comments.